### ISSUE TITLE
Batch insert using map seems to still not work

### ISSUE THREAD

**Issue by saantiaguilera** [jmoiron/sqlx#701]
Hi, I'm using the master branch since it has the fix of batch inserts on named arguments. Still, when using a map it seems to still fail:

```go
_, err := tx.NamedExec(
    "INSERT INTO state (state_id, name, created_at) VALUES (:id, :name, :time) ON DUPLICATE KEY UPDATE state_id=state_id",
    []map[string]interface{}{
        {"id": "MX-MEX", "name": "Ciudad de México", "time": time.Now()},
	{"id": "BR-SP", "name": "São Paulo", "time": time.Now()},
    },
)
```

Seems to fail with error: `sql: expected 3 arguments, got 6`

Any idea why?

**Comment by hyperhtml** [comment]
Nothing to add here, but I just ran into this exact issue as well.    

I can get the above query to run if you omit `ON DUPLICATE KEY UPDATE state_id=state_id`.   However, I also need that same clause in my query.

**Comment by alexshtin** [comment]
I have same issue. Pinned to `1.2.0` for now.

**Comment by jmoiron** [comment]
Hesitantly going to call this fixed with #734, which is tagged with `1.3.4`
